site/profile/manifests/userportal.pp

```diff
@@ -0,0 +1,121 @@
+  package {['python3-virtualenv', 'python38', 'python38-devel']: }
+  package {['openldap-devel', 'gcc', 'mariadb-devel']: }
+
+  # Using python3.8 with gunicorn
+  exec { 'create virtualenv':
+    command => '/usr/bin/virtualenv-3 --python="/usr/bin/python3.8" /var/www/userportal-env',
+    unless  => '/usr/bin/test -d /var/www/userportal-env',
+    require => [Package['python3-virtualenv'], Package['python38']],
+  }
+
+  file { '/var/www/userportal/':
+    ensure => 'directory',
+    owner  => 'apache',
+    group  => 'apache',
+  }
+  -> vcsrepo { '/var/www/userportal/':
+    ensure   => present,
+    provider => git,
+    source   => 'https://github.com/guilbaults/TrailblazingTurtle.git',
+    revision => '7940ae14891a60d18afd1d9d009dada044512b0f',
+    user     => 'apache',
+    notify   => [Service['httpd'], Service['gunicorn']],
+  }
+  -> file { '/var/www/userportal/userportal/settings.py':
+    show_diff => false,
+    content   => epp('profile/userportal/settings.py',
+      {
+        'password'     => $password,
+        'cluster_name' => lookup('profile::slurm::base::cluster_name'),
+        'secret_key'   => fqdn_rand_string(32, undef, $password),
+        'domain_name'  => $domain_name,
+      }
+    ),
+    notify    => Service['gunicorn'],
+  }
+  -> file { '/var/www/userportal/userportal/common.py':
+    source => 'file:/var/www/userportal/example/common.py',
+    notify => [Service['httpd'], Service['gunicorn']],
+  }
+  -> exec { 'pip install -r':
+    command => '/var/www/userportal-env/bin/pip3 install -r /var/www/userportal/requirements.txt',
+    require => [Exec['create virtualenv'], Package['python38-devel'], Package['gcc']],
+  }
+
+  # Need to use this fork to manage is_staff correctly
+  # https://github.com/enervee/django-freeipa-auth/pull/9
+  -> exec { 'pip install django-freeipa-auth':
+    command => '/var/www/userportal-env/bin/pip3 install git+https://github.com/88Ocelot/django-freeipa-auth.git',
+    unless  => '/var/www/userportal-env/bin/pip3 freeze | /usr/bin/grep django-freeipa-auth',
+    require => [Exec['create virtualenv']],
+  }
+
+  file { '/var/www/userportal-static':
+    ensure => 'directory',
+    owner  => 'apache',
+    group  => 'apache',
+  }
+
+  file { '/etc/httpd/conf.d/userportal.conf':
+    content => epp('profile/userportal/userportal.conf.epp'),
+    notify  => Service['httpd'],
+  }
+
+  file { '/etc/systemd/system/gunicorn.service':
+    mode    => '0755',
+    content => '[Unit]
+Description=gunicorn daemon
+After=network.target
+
+[Service]
+User=apache
+Group=apache
+RuntimeDirectory=gunicorn
+WorkingDirectory=/var/www/userportal/
+ExecStart=/var/www/userportal-env/bin/gunicorn --bind 127.0.0.1:8001 --workers 2 --timeout 90 userportal.wsgi
+ExecReload=/bin/kill -s HUP $MAINPID
+KillMode=mixed
+TimeoutStopSec=5
+PrivateTmp=true
+
+[Install]
+WantedBy=multi-user.target',
+    notify  => Service['gunicorn'],
+  }
+
+  service { 'gunicorn':
```

Comment:
We would want a less generic name for that service, given there could be more than just the user-portal's gunicorn process running.